osquery/events/linux/syslog.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <fcntl.h>
+#include <grp.h>
+#include <sys/file.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <istream>
+#include <string>
+
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/tokenizer.hpp>
+
+#include <osquery/logger.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/events/linux/syslog.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+FLAG(string,
+     syslog_pipe_path,
+     "/var/osquery/syslog_pipe",
+     "Path to the named pipe used for forwarding rsyslog events");
+
+REGISTER(SyslogEventPublisher, "event_publisher", "syslog");
+
+// rsyslog needs read/write access, osquery process needs read access
+const mode_t kPipeMode = 0460;
+const std::string kPipeGroupName = "syslog";
+const char* kTimeFormat = "%Y-%m-%dT%H:%M:%S";
```

Comment:
Yes, there will be a wiki page explaining how to configure `rsyslog` properly for this.